lib/internal/webstreams/transformstream.js

```diff
@@ -119,10 +125,13 @@ class TransformStream {
    * @param {QueuingStrategy} [readableStrategy]
    */
   constructor(
-    transformer = null,
+    transformer = {},
```

Comment:
```suggestion
    transformer = { __proto__: null },
```

Reply:
Hmm, it looks like there's some inconsistencies between implementations here. 👀

Have a look at this snippet:
```javascript
Object.prototype.start = () => console.log("start called");
rs = new ReadableStream();
```
Chrome does go through `Object.prototype` when accessing the `source.start` method:
![image](https://github.com/nodejs/node/assets/649348/3caa1c6a-52a1-469b-8314-7f232562f207)

However, Firefox doesn't:
![image](https://github.com/nodejs/node/assets/649348/2d949cc2-a297-44a6-964e-213b66513f02)

I'll check the reference implementation...

Reply:
Interesting - we should do whatever the web spec says to do here.

Reply:
This keeps getting more interesting! 😅

[This is the relevant part of the reference implementation.](https://github.com/whatwg/streams/blob/d58a68c9de222562532b49303f9078ca3103c1b8/reference-implementation/lib/ReadableStream-impl.js#L15-L18) For `new ReadableStream()` (without any arguments), we have `underlyingSource === null` before calling `UnderlyingSource.convert`. That conversion function is auto-generated by webidl2js, and looks like this:
```javascript
  {
    const key = "start";
    let value = obj === undefined || obj === null ? undefined : obj[key];
    if (value !== undefined) {
      value = UnderlyingSourceStartCallback.convert(globalObject, value, {
        context: context + " has member 'start' that"
      });

      ret[key] = value;
    }
  }
```
Since `obj === null`, we never even try to access `obj.start`.

However, if you call `new ReadableStream({})` (with an empty object), the reference implementation *does* access `Object.prototype.start`. And indeed, both Chrome and Firefox log "start called" for this snippet:
```javascript
Object.prototype.start = () => console.log("start called");
rs = new ReadableStream({});
// -> start called
```

I believe the reference implementation and Firefox are behaving correctly, and Chrome isn't. So I'll make Node use a `null` prototype too. 👍

Either way, I should really make a WPT test for this. 😅